### PR TITLE
Docs navigation improvements, docs wording

### PR DESCRIPTION
**Pull request by ppalaga**
@jamesnetherton could you please throw an eye on the spelling?

### CODE REVIEW

File: docs/modules/ROOT/pages/create-new-extension.adoc [hunk 1]
```diff
@@ -0,0 +1,79 @@
+[[create-a-new-extension]]
+= Create a new extension
+
+1. You should know link:#how-to-build[how to build].
+
+2. Go through the https://quarkus.io/guides/extension-authors-guide[Quarkus extension author's guide] to get an idea
+   what is expecting you.
```
Comment (jamesnetherton) [inline docs/modules/ROOT/pages/create-new-extension.adoc:7]: It's better as: `to get an idea of what is expected from you`.

File: docs/modules/ROOT/pages/create-new-extension.adoc [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+* It creates three new Maven modules under the `extensions` directory: `camel-quarkus-foo-abc-parent`, `camel-quarkus-foo-abc`
+  (a.k.a. the runtime module) and `camel-quarkus-foo-abc-deployment`.
+* These three modules are linked where necessary:
+** `camel-quarkus-foo-abc-parent` is added to the `<modules>` of `camel-quarkus-extensions`
+** `camel-quarkus-foo-abc` is added to the `<dependencyManagement>` of the runtime BOM (Bill of Materials) `poms/bom/pom.xml`
+** `camel-quarkus-foo-abc-deployment` is added to the `<dependencyManagement>` of the deployment BOM (Bill of Materials) `poms/bom-deployment/pom.xml`
+* It creates a basic `FooAbcProcessor` class in the deployment module.
+* It also creates a stub of an integration test module under `integration-tests/foo-abc`.
++
+A compilation performed immediately after generating the modules should pass flawlessly but running the tests will fail
```
Comment (jamesnetherton) [inline docs/modules/ROOT/pages/create-new-extension.adoc:38]: Maybe better to start the sentence as: `Compilation is performed immediately after...`

File: docs/modules/ROOT/pages/create-new-extension.adoc [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+because the test project needs to get finished. You need to build `poms/bom` and `poms/bom-deployment` one time first.
+
+6. Review the generated
+   `extensions/foo-abc/runtime/src/main/resources/META-INF/quarkus-extension.yaml` file. If you
+   see improper description or keyword,
```
Comment (jamesnetherton) [inline docs/modules/ROOT/pages/create-new-extension.adoc:43]: Better as: `see an improper description or keyword`.

File: docs/modules/ROOT/pages/create-new-extension.adoc [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+   https://issues.apache.org/jira/secure/CreateIssue!default.jspa[file a new Camel issue] and ask to fix the metadata
+   for the given Camel component.
+7. Review the dependencies in the generate runtime and deployment modules. In case the given library is supported by
```
Comment (jamesnetherton) [inline docs/modules/ROOT/pages/create-new-extension.adoc:46]: `generated` instead of `generate`

**Comment by ppalaga** [comment]
Thanks @jamesnetherton, all suggestions accepted.

**Pull request merged**